types/search-grounding.ts

```diff
@@ -0,0 +1,165 @@
+/**
+ * @license
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { DynamicRetrievalMode } from "./enums";
+
+/**
+ * Retrieval tool that is powered by Google search.
+ * @public
+ */
+export declare interface GoogleSearchRetrievalTool {
+  /**
+   * Google search retrival tool config.
+   */
+  googleSearchRetrieval?: GoogleSearchRetrieval;
+}
+
+/**
+ * Retrieval tool that is powered by Google search.
```

Comment:
Mark all public interfaces with `@public` (don't need to worry about tagging the properties, they will automatically be included)

Reply:
Done